src/main/java/christmas/domain/utility/Parser.java

```diff
@@ -0,0 +1,113 @@
+package christmas.domain.utility;
+
+import christmas.domain.consumer.constants.Menu;
+import christmas.exception.BusinessException;
+import christmas.exception.ExceptionHandler;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static christmas.exception.ErrorCode.INVALID_ORDER;
+
+public class Parser {
+    private static final int POSITIVE_NUMBER_MINIMUM_RANGE = 1;
+    private static final String DELIMITER = ",";
+    private static final String HYPHEN = "-";
+    private static final Pattern REGEX_PATTERN = Pattern.compile("^[가-힣]+-\\d{1,20}$");
```

Comment:
```suggestion
    private static final Pattern REGEX_PATTERN = Pattern.compile("^[가-힣]+-\\d+$");
```
혹시 REGEX의 \\d{1, 20}가 `1~20의 정수`를 받는다는 의도셨다면, 바로잡아드리고싶네요.
다음의 부분은 `1에서 20개 사이의 digit` 이라는 뜻입니다! 다른 메서드에서 Integer.parseInt 이후에 검증하는게 좋을 것 같아요.

Reply:
이번 과제에서 정규표현식을 처음으로 사용해보았는데요..
사용 과정에서 조금 문제가 있었네요!

말씀주신 내용을 바탕으로, 다시 공부해보니
1~20자리의 숫자를 보장하는 정규표현식이라는 것을 확인하게 되었습니다!!
물론, 총 주문횟수에서 이를 검증하면서 오류가 발생하지는 않지만
개별 주문이 20개가 넘으면 에러를 던지려고 기존에 설계했던 로직 자체는
조금 의도하는 바와 다르게 동작했던 것 같습니다!!

날카롭고 예리한 리뷰 감사드립니다 :)